cmake/utils.cmake

```diff
@@ -150,7 +147,7 @@ function(mayaUsd_install_rpath rpathRef NAME)
 endfunction()
 
 function(mayaUsd_promoteMayaUsdHeader)
-    set(srcFile ${CMAKE_CURRENT_SOURCE_DIR}/base/mayaUsd.h.src)
+    set(srcFile ${CMAKE_CURRENT_SOURCE_DIR}/mayaUsd.h.src)
     set(dstFile ${CMAKE_BINARY_DIR}/include/mayaUsd/mayaUsd.h)
     if (NOT EXISTS ${dstFile})
         message(STATUS "promoting: " ${srcFile})
```

Comment:
I'm not really sure why there's a utility function for this. Seems like its contents should just go in the `lib` (or `lib/base`?) CMakeLists.txt?

Reply:
I can understand why this could be confusing. 

This utility basically calls cmake "configure_file" function to copy an `input `file to an `output`  file and substitutes variable values referenced as @VAR@ or ${VAR} in the input file content.

```
input =  mayaUsd.h.src
output = include/mayaUsd/mayaUsd.h
```
![image](https://user-images.githubusercontent.com/48299423/75990244-98e48180-5ec2-11ea-9e58-51a5c749915c.png)

The reason `base/` is removed is because every sub-directories under "lib" is now getting it's own `CMakeLists.txt` and we no longer need to have hard-coded paths in utilities anymore.

Reply:
Oh, sorry if I was unclear. I think I get what the function does. My gripe was more that I think as a function, this only gets called in one place, doesn't it? I would think we only install the `mayaUsd.h` file once. Can/should we just put this code directly in the lib/base/CMakeLists.txt?

Reply:
Correct, this function is only called once for copying mayaUsd.h.src which is in lib/base/CMakeLists.txt

You have a point, this is now directly called in lib/base/CMakeLists.txt. 

See 71d2a8bf1